client/src/commands/testSmartContractCommand.ts

```diff
@@ -74,6 +74,27 @@ export async function testSmartContract(chaincode?: ChainCodeTreeItem): Promise<
     }
     console.log('testSmartContractCommand: chaincode to generate tests for is: ' + chaincodeLabel);
 
+    // Only generate the test file if the smart contract is open in the workspace
+    const packageJSONSearch: Array<vscode.Uri> = await vscode.workspace.findFiles('**/package.json', '**/node_modules/**', 10);
```

Comment:
Can you work out the number of folders so you can just look in all the top level folders in the workspace